src/scheduler.rs

```diff
@@ -0,0 +1,32 @@
+pub mod task_context;
+pub mod task_stack;
+
+use crate::interupts::ExceptionContext;
+
+pub fn yeet() {
+    unsafe {
+        llvm_asm!("svc 0");
+    }
+}
+
+#[no_mangle]
+#[inline(never)]
+pub extern "C" fn foo() {
+    crate::println!("BEHOLD! SECOND TASK");
+    loop {}
+}
+
+pub fn sample_change_task(_e: &mut ExceptionContext) -> &mut ExceptionContext {
+    let task = match task_context::TaskContext::new(foo) {
+        Ok(t) => t,
+        Err(_) => {
+            crate::println!(">>>>>> ERROR CREATING TASK CONTEXT");
```

Comment:
Yo can give more reasons in error msg
`
 Err(e) => {
            crate::println!(">>>>>> ERROR CREATING TASK CONTEXT: {:?}", e);
}`
If you derive Debug on error Type